e2e/wdio/core/tests/busy-indicator.e2e-spec.ts

```diff
@@ -0,0 +1,170 @@
+            expect(busyIndicatorPage.compareWithBaseline('busy-indicator', componentExample)).toBeLessThan(1);
+        });
+
+        it('Check disable button hover state', () => {
+            scrollIntoView(enableDisableButton);
+            checkButtonHoverState(enableDisableButton, disableButtonExample + disableButtonHover + '-', 'disable-button');
+        });
+
+        it('Check disable button active state', () => {
+            scrollIntoView(enableDisableButton);
+            checkButtonActiveState(enableDisableButton, disableButtonExample + disableButtonActive + '-', 'disable-button');
+        });
+
+        it('Check disable button focus state', () => {
+            scrollIntoView(enableDisableButton);
+            checkButtonFocusState(enableDisableButton, disableButtonExample + disableButtonFocus + '-', 'disable-button');
+        });
+
+        it('Check save button hover state', () => {
+            scrollIntoView(saveButton);
+            click(enableDisableButton);
+            checkButtonHoverState(saveButton, saveButtonExample + saveButtonHover + '-', 'save-button');
+        });
+
+        it('Check save button active state', () => {
+            scrollIntoView(saveButton);
+            click(enableDisableButton);
+            checkButtonActiveState(saveButton, saveButtonExample + saveButtonActive + '-', 'save-button');
+        });
+
+        it('Check save button focus state', () => {
+            scrollIntoView(saveButton);
+            click(enableDisableButton);
+            checkButtonFocusState(saveButton, saveButtonExample + saveButtonFocus + '-', 'save-button');
+        });
+
+        it('Check enable loading button hover state', () => {
+            scrollIntoView(enableDisableButton);
+            click(enableDisableButton);
+            checkButtonHoverState(enableDisableButton, enableLoadingButtonExample + enableLoadingButtonHover + '-', 'save-button');
+        });
+
+        it('Check enable loading button active state', () => {
+            scrollIntoView(enableDisableButton);
+            click(enableDisableButton);
+            checkButtonActiveState(enableDisableButton, enableLoadingButtonExample + enableLoadingButtonActive + '-', 'save-button');
+        });
+
+        it('Check enable loading button focus state', () => {
+            scrollIntoView(enableDisableButton);
+            click(enableDisableButton);
+            checkButtonFocusState(enableDisableButton, enableLoadingButtonExample + enableLoadingButtonFocus + '-', 'save-button');
+        });
+
+        it('Check indicator for block focus state', () => {
+            scrollIntoView(enableDisableButton);
+            checkButtonFocusState(indicatorBlockWrapper, indicatorBlockExample + indicatorBlockFocus + '-', 'indicator-block');
+        });
+    });
+});
+
+function checkButtonHoverState(selector: string, tag: string, btnName: string, index: number = 0): void {
```

Comment:
DRY principle: if you're using the same functions in multiple tests, then let's move these to the assertion helper file so they can be reused.